### PR TITLE
Disable both solution crawlers if option is disabled

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1753274

Previously, we had an option for this, but it required all *consumers* to check it and not call into solution-crawler if it was off.  Now, the checks just go into solution-crawler (and the unit-testing version) and all the consumers don't care and just call into them.

### CODE REVIEW

File: src/EditorFeatures/Core/Interactive/InteractiveWorkspace.cs [hunk 1]
```diff
@@ -16,20 +16,17 @@ internal partial class InteractiveWorkspace : Workspace
...
-                Services.GetRequiredService<ISolutionCrawlerRegistrationService>().Register(this);
-            }
+            DiagnosticProvider.Enable(this);
         }
 
         protected override void Dispose(bool finalize)
         {
             // workspace is going away. unregister this workspace from work coordinator
-            Services.GetRequiredService<ISolutionCrawlerRegistrationService>().Unregister(this, blockingShutdown: true);
+            DiagnosticProvider.Disable(this);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Interactive/InteractiveWorkspace.cs:29]:
```suggestion
            DiagnosticProvider.Disable(this, blockingShutdown: true);
```

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingSolutionCrawlerRegistrationService.cs [hunk 2]
```diff
@@ -41,20 +43,25 @@ internal partial class UnitTestingSolutionCrawlerRegistrationService : IUnitTest
...
         }
 
         /// <summary>
         /// make sure solution cralwer is registered for the given workspace.
         /// </summary>
-        public IUnitTestingWorkCoordinator Register(Solution solution)
+        public IUnitTestingWorkCoordinator? Register(Solution solution)
         {
+            if (!_globalOptionService.GetOption(SolutionCrawlerRegistrationService.EnableSolutionCrawler))
+                return null;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingSolutionCrawlerRegistrationService.cs:63]: this is hte actual service.  so it checks the option and bails if not on.

File: src/EditorFeatures/Core/Interactive/InteractiveWorkspace.cs [hunk 3]
```diff
@@ -16,20 +16,17 @@ internal partial class InteractiveWorkspace : Workspace
...
-        internal InteractiveWorkspace(HostServices hostServices, IGlobalOptionService globalOptions)
+        internal InteractiveWorkspace(HostServices hostServices)
             : base(hostServices, WorkspaceKind.Interactive)
         {
             // register work coordinator for this workspace
-            if (globalOptions.GetOption(SolutionCrawlerRegistrationService.EnableSolutionCrawler))
-            {
-                Services.GetRequiredService<ISolutionCrawlerRegistrationService>().Register(this);
-            }
+            DiagnosticProvider.Enable(this);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Interactive/InteractiveWorkspace.cs:23]: consumers do not check anymore.  they just call into service to enable themselves. the service decides if it needs to do anything or not.

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingSolutionCrawlerRegistrationService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingSolutionCrawlerRegistrationService.cs:63]: also, this is hte unit testing side, which wasn't checking the option.

File: src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerRegistrationService.cs [hunk 4]
```diff
@@ -76,6 +82,9 @@ public void EnsureRegistration(Workspace workspace, bool initializeLazily)
         {
             Contract.ThrowIfNull(workspace.Kind);
 
+            if (!_globalOptionService.GetOption(SolutionCrawlerRegistrationService.EnableSolutionCrawler))
+                return;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerRegistrationService.cs:86]: this is hte actual service.  so it checks the option and bails if not on.

**Review by tmat — CHANGES_REQUESTED** [review]
:clock1:

**Pull request closed without merging**